onnxruntime/core/framework/cloud_invoker.cc

```diff
@@ -33,6 +36,174 @@ CloudEndPointInvoker::CloudEndPointInvoker(const CloudEndPointConfig& config,
   }
 }
 
+class CurlGlobal {
+ public:
+  static void Init() {
+    static CurlGlobal curl_global;
+  }
+
+ private:
+  CurlGlobal() {
+    // Thread-safety is a must since curl might also be initialized in triton client.
+    const auto* info = curl_version_info(CURLVERSION_NOW);
+    ORT_ENFORCE(info->features & CURL_VERSION_THREADSAFE, "curl global init not thread-safe, need to upgrade curl version!");
+    ORT_ENFORCE(curl_global_init(CURL_GLOBAL_DEFAULT) == CURLE_OK, "Failed to initialize curl global env!");
```

Comment:
Even through curl_global_init is thread-safe, we still should not call it from any global variable's constructor since at that time the static variables in libcurl itself might have not been initialized. So I think the best way to ensure is to put this call in OrtEnv's constructor.

Reply:
Don't get it, how come they are not initialized while OrtEnv constructor could avoid that?

Reply:
Talked with Randy offline. The most difficult part is:  this file also uses TritonClient lib which also has similar code that initialize and deinit curl. I don't see an easy way to coordinate them together.  Will leave the discussion  later.